.github/workflows/system-tests.yml

```diff
@@ -15,12 +15,12 @@ jobs:
     outputs:
       outcome: ${{ steps.run_needed.outcome }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - id: run_needed
         name: Check if run is needed
         run:  |
-          git fetch origin ${{ github.event.pull_request.base.sha || 'main' }}
-          export PATHS=$(git diff --name-only HEAD ${{ github.event.pull_request.base.sha }})
+          git fetch origin ${{ github.event.pull_request.base.sha || github.sha }}
```

Comment:
Just for my own context/clarification, `github.event.pull_request.base.sha` is checking for the SHA hash of the base commit of the PR, and the `github.sha` is the SHA hash of the commit that was most recently pushed and triggered the workflow?

Reply:
Yup, that's right. Here's the documentation https://docs.github.com/en/actions/learn-github-actions/contexts#github-context
The main reason to make this change is that the PR base SHA doesn't exist when this action runs on `main`.